src/basic/compress.c

```diff
@@ -65,6 +65,16 @@ static const char* const compression_table[_COMPRESSION_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(compression, Compression);
 
+bool compression_supported(Compression c) {
+        static const unsigned supported =
+                (1U << COMPRESSION_NONE) |
+                (1U << COMPRESSION_XZ) * HAVE_XZ |
+                (1U << COMPRESSION_LZ4) * HAVE_LZ4 |
+                (1U << COMPRESSION_ZSTD) * HAVE_ZSTD;
```

Comment:
At `-O0` this generates quite a long function, because `supported` is not inlined. But at `-O2` it works nicely:
```
000000000010daa0 <compression_supported>:
                (1U << COMPRESSION_NONE) |
                (1U << COMPRESSION_XZ) * HAVE_XZ |
                (1U << COMPRESSION_LZ4) * HAVE_LZ4 |
                (1U << COMPRESSION_ZSTD) * HAVE_ZSTD;

        return c >= 0 && c < _COMPRESSION_MAX && FLAGS_SET(supported, 1U << c);
  10daa0:       83 ff 03                cmp    $0x3,%edi
  10daa3:       0f 96 c0                setbe  %al
}
  10daa6:       c3                      ret
  10daa7:       66 0f 1f 84 00 00 00    nopw   0x0(%rax,%rax,1)
  10daae:       00 00 
```
:+1: